packages/core/src/hooks/useRawLogs.ts

```diff
@@ -1,10 +1,75 @@
-import { useEffect, useMemo, useState } from 'react'
+import { useEffect, useMemo, useRef, useState } from 'react'
 import { useEthers } from './useEthers'
 import { useReadonlyNetworks } from '../providers/network/readonlyNetworks'
-import { useBlockNumbers, useBlockNumber } from '../hooks'
+import { useBlockNumbers, useBlockNumber, useConfig } from '../hooks'
 import { QueryParams } from '../constants/type/QueryParams'
 import type { Filter, FilterByBlockHash, Log } from '@ethersproject/abstract-provider'
 import { Falsy } from '../model/types'
+import { ChainId } from '../constants'
+
+function deepEqual(obj1: any, obj2: any) {
+  if (obj1 === obj2)
+    // it's just the same object. No need to compare.
+    return true
+
+  if (obj1 == null) return obj1 == null
+
+  if (obj2 == null) return false
+
+  if (isPrimitive(obj1) && isPrimitive(obj2))
+    // compare primitives
+    return obj1 === obj2
+
+  if (Object.keys(obj1).length !== Object.keys(obj2).length) return false
+
+  // compare objects with same number of keys
+  for (const key in obj1) {
+    if (!(key in obj2)) return false //other object doesn't have this prop
+    if (!deepEqual(obj1[key], obj2[key])) return false
+  }
+
+  return true
+}
+
+function isPrimitive(obj: any) {
+  return obj !== Object(obj)
+}
+
+function useResolvedFilter(
+  filter: Filter | FilterByBlockHash | Promise<Filter | FilterByBlockHash> | Falsy
+): Filter | FilterByBlockHash | Falsy {
+  const [resolvedFilter, setResolvedFilter] = useState<Filter | FilterByBlockHash | Falsy>(
+    filter instanceof Promise ? undefined : filter
+  )
+
+  useEffect(() => {
+    let active = true // Flag to prevent setting state after unmount
+
+    const resolveFilter = async () => {
+      let _filter: Filter | FilterByBlockHash | Falsy = undefined
+
+      if (filter instanceof Promise) {
+        _filter = await filter
+      } else {
+        _filter = filter
+      }
```

Comment:
```suggestion
      let _filter: Filter | FilterByBlockHash | Falsy = await filter
```
If you await non promise value it will just return it as value otherwise it will resolve promise

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#return_value

Reply:
Clean